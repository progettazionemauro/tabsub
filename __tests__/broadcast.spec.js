import mockBroadcastChannel from '../__setup__/broadcast';
import tabsub from '../src/index';

let restoreBroadcast;

beforeAll(() => (restoreBroadcast = mockBroadcastChannel()));
afterAll(() => restoreBroadcast());

it('is used as default strategy', () => {
  const p = tabsub('channel');
  expect(p).toMatchSnapshot();
});

it('posts messages to subscribers', () => {
  const subscriberMockCb = jest.fn(msg => msg);
  const p = tabsub('channel1');
  const s = tabsub('channel1');
  s.on(subscriberMockCb);

  p.post('A msg');
  p.post('Another one');
  p.post('A third one');

  expect(subscriberMockCb.mock.calls).toMatchSnapshot();
});

it('posts messages with different types', () => {
  const subscriberMockCb = jest.fn(msg => msg);
  const p = tabsub('channel2');
  const s = tabsub('channel2');
  s.on(subscriberMockCb);

  p.post('A msg');
  p.post({foo: 'bar'});
  p.post({baz: 1});
  p.post(1);
  p.post();
  p.post(['lovely', 'quz', 1, 0.4]);

  expect(subscriberMockCb.mock.calls).toMatchSnapshot();
});

it('stops and starts the messages on demand', () => {
  const subscriberMockCb = jest.fn(msg => msg);
  const p = tabsub('channel3');
  const s = tabsub('channel3');
  s.on(subscriberMockCb);

  p.post('A msg');
  s.stop();

  p.post('this will get ignored from the subscriber');
  p.post('And this one');
  s.start();
  p.post('Keep them coming');
  expect(subscriberMockCb.mock.calls).toMatchSnapshot();
});
