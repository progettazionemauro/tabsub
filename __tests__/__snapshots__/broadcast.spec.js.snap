// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is used as default strategy 1`] = `
Object {
  "close": [Function],
  "name": "channel",
  "on": [Function],
  "post": [Function],
  "start": [Function],
  "stop": [Function],
}
`;

exports[`posts messages to subscribers 1`] = `
Array [
  Array [
    "A msg",
  ],
  Array [
    "Another one",
  ],
  Array [
    "A third one",
  ],
]
`;

exports[`posts messages with different types 1`] = `
Array [
  Array [
    "A msg",
  ],
  Array [
    Object {
      "foo": "bar",
    },
  ],
  Array [
    Object {
      "baz": 1,
    },
  ],
  Array [
    1,
  ],
  Array [
    null,
  ],
  Array [
    Array [
      "lovely",
      "quz",
      1,
      0.4,
    ],
  ],
]
`;

exports[`stops and starts the messages on demand 1`] = `
Array [
  Array [
    "A msg",
  ],
  Array [
    "Keep them coming",
  ],
]
`;
